vm/src/obj/objbyteinner.rs

```diff
@@ -94,7 +98,148 @@ impl PyByteInner {
             }
         }
     }
+}
+
+#[derive(FromArgs)]
+pub struct ByteInnerFindOptions {
+    #[pyarg(positional_only, optional = false)]
+    sub: PyObjectRef,
+    #[pyarg(positional_only, optional = true)]
+    start: OptionalArg<PyObjectRef>,
+    #[pyarg(positional_only, optional = true)]
+    end: OptionalArg<PyObjectRef>,
+}
+
+impl ByteInnerFindOptions {
+    pub fn get_value(
+        self,
+        elements: &[u8],
+        vm: &VirtualMachine,
+    ) -> PyResult<(Vec<u8>, Range<usize>)> {
+        let sub = match try_as_bytes_like(&self.sub) {
```

Comment:
There are many cases where we can get "bytes like" or `int`. Maybe consider making some sort of abstraction for this case?